### DISCUSSION TITLE
Is there any fish bass in nushell? Import environment from bash script execution.

### DISCUSSION THREAD

**Discussion by Abdillah** [nushell/nushell#11505] [Q&A] [answered]
Fish shell [bass](https://github.com/edc/bass/) can execute or source bash script and applying the environment to the running Fish shell.

My specific goal is to read NixOS `/etc/set-environment` which contains Bash environment definition on startup. It might be an arbitrary script like nvm.

Is there any project like this in Nushell?

**Comment by fdncred** [comment]
i've seen people parse `export key=value` env vars from a file, but that's about it.

**Reply by Abdillah** [reply]
Thanks for sharing.
Bass has python side, maybe I can make simple wrapper for Nushell to parse the resulting `key=value`.

**Comment by Abdillah** [comment, marked as answer]
I found this plugin, but can't wrap my head around it.

> https://github.com/tesujimath/nu_plugin_bash_env

If I look at the examples, it is indeed loading bash environment from `anybashexportfile.env` and make as JSON to be loaded with `load-env`.

**Reply by Abdillah** [reply]
It is!

Just copy the script to `$NU_HOME/plugins/nu_plugin_bash_env` and `bash-env` command will be available. Call any bash file with it, don't forget to pipe to `load-env`.
